lighthouse/helpers/plates.py

```diff
@@ -263,6 +263,124 @@ def send_to_ss_heron_plates(body: Dict[str, Any]) -> requests.Response:
+
+    # Validate plate purpose is as expected
+    # plate_purpose = None
+    if (included := json["included"]) is not None:
+        purposes = [elem for elem in included if elem["type"] == "purposes"]
+
+        # Validate only one plate pupose exists
+        # TODO (DPL-572): check if is this needed? Could there ever be more than one purpose?
+        if len(purposes) != 1:
+            return {"data": None, "error": f"There should only be one purpose for barcode '{barcode}'"}
+
+        purpose_name = purposes[0]["attributes"]["name"]
+
+        if purpose_name != BIOSCAN_PLATE_PURPOSE:
+            return {"data": None, "error": f"Incorrect purpose '{purpose_name}' for barcode '{barcode}'"}
+
+        # Validate samples exists for plate
+        samples = [elem for elem in included if elem["type"] == "samples"]
+        if len(samples) == 0:
+            return {"data": None, "error": f"There are no samples for barcode '{barcode}'"}
+
+    # Instantiate wrapper class
+    control_locations = ControlLocations(included)
+
+    # Perform data processing, to retrieve well controls
+    control_info = control_locations.get_control_locations()
+    # { well_position: control_type } e.g.  {'A1': 'pcr positive', 'B1': 'pcr negative'}
+
+    # Get the well position, for the +ve and -ve control types
+    positive_control_position = [k for (k, v) in control_info.items() if v == "pcr positive"]
+    negative_control_position = [k for (k, v) in control_info.items() if v == "pcr negative"]
+
+    # Validate only one positive and one negative controls exist
+    if len(positive_control_position) > 1 or len(negative_control_position) > 1:
+        return {
+            "data": None,
+            "error": f"There should be only one positive and one negative control for barcode '{barcode}'",
+        }
+
+    # Validate both controls exist
+    if len(positive_control_position) != 1 or len(negative_control_position) != 1:
+        return {"data": None, "error": f"Missing positive or negative control for barcode '{barcode}'"}
+
+    # Convert data into expected format for Beckman
+    # This response was defined by the Beckman SAT
+    locations = {
+        "barcode": barcode,
+        "positive_control": positive_control_position[0],
+        "negative_control": negative_control_position[0],
+    }
+
+    return {"data": locations, "error": None}
+
+
+def get_from_ss_plates_samples_info(plate_barcode: str) -> requests.Response:
+    ss_url: str = f"{app.config['SS_URL']}/api/v2/labware"
+
+    try:
+        params = {"filter[barcode]": plate_barcode, "include": "purpose,receptacles.aliquots.sample"}
+        response = requests.get(f"{ss_url}", params=params)
+
+        logger.debug(f"Response status code: {response.status_code}")
+
+        assert "data" in response.json(), f"Expected 'data' in response: {response.json()}"
+
+        return response
+    except requests.ConnectionError:
```

Comment:
For any errors returned, do we always return the error in the format expected by Beckman, as in the Postman SAT testing?

Reply:
From the tests looks like it does.

Reply:
A couple of 500 errors were not in the expected format. I converted them to expected error format. Except for the response code, they look the same. 

However, there may be others that are unlikely to happen may have a different format. I am trying to find those cases. One of them would be because of a badly formatted post data. The problem is that it comes from the JSON decoding library.